src/mca/gds/ds12/gds_dstore.c

```diff
@@ -63,60 +63,216 @@
 #define ESH_KV_SIZE(addr)                                   \
 __extension__ ({                                            \
     size_t sz;                                              \
-    memcpy(&sz, addr, sizeof(size_t));                      \
+    if (PMIX_PROC_IS_V1(_client_peer())) {                  \
```

Comment:
@karasevb We discussed that those will be a function pointers to avoid extra "ifs" all around the code. Please fix.